### DISCUSSION TITLE
Undefined body from server request?

### DISCUSSION THREAD

**Discussion by stur86** [mswjs/msw#1240] [Q&A]
I'm asking here since I'm not sure if this is worth an issue or just me doing something wrong, but I'm trying to use `startServer` for a test with a React component, where I need an API to return a byte stream. I'm doing this:

```js
const server = setupServer(
    rest.get("http://stream", (req, res, ctx) => {
      const data = new Uint8Array([1,2,3,4]);
      return res(
        ctx.set('Content-Length', data.byteLength.toString()),
        ctx.set('Content-Type', 'application/octet-stream'),
        ctx.body(data)
      );
    })
);

beforeAll(() => server.listen());
afterAll(() => server.close());
```

and when I perform a fetch request I get a response with status 200, but the body is undefined. It doesn't matter what I try, I tried also using a string as a body and remove the headers, still nothing. Is this a mistake in how I'm using it or a bug? I've seen something similar used to happen due to a problem with `@msw/interceptors` but I thought that had been fixed.

**Comment by kettanaito** [comment]
Hi, @stur86.

Could you please share how you're reading that response? A reproduction repository would be ideal. You can clone one of [our examples](https://github.com/mswjs/examples/tree/master/examples) or [our CodeSandbox template](https://codesandbox.io/s/msw-react-xx1c8) to create a reproduction repository much faster.

**Comment by stur86** [comment]
Hi, 

I'm trying but I'm having a hard time getting to reproduce the bug. I can't share the original code since it's proprietary, but the test examples have plenty of differences - even the version of React is still very behind (it's 16.7 while I'm using 18.0). I'll probably have to make my own repository to get to see this working. Any clues as to what packages or settings might be critical to this functionality?

**Reply by kettanaito** [reply]
I think it's important to use the same versions of third-party request libraries you may be using (react-query, axios, if any). The first place I'd look at would be how you're reading that stream.

Speaking of React, I don't think it's the culprit but I understand that you wish to match your environment as close as possible, which is a good intention if we wish to reproduce this issue. You can fork that sandbox I've linked and update packages, they don't have to be exact!

**Reply by stur86** [reply]
Ok, will look into it. I was only using `fetch` directly really, so I don't think any library was involved there. When I tried it on your sandbox though it seemed to work so I have to wonder if there's some weird side effect from something else.

**Comment by stur86** [comment]
So this is the smallest example I could create using CRA:

https://github.com/stur86/msw-bug-fetch-report

Same Typescript version as my project (later ones are incompatible with MSW apparently). The relevant example is found in `myHook.test.tsx`.

**Reply by kettanaito** [reply]
I think responding with streams in Node.js is not supported (not sure how that works in the browser, if anyhow). I've opened #1284 to tackle different response types.
